src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -2053,24 +2057,69 @@ UpdateShardPlacement(PlacementUpdateEvent *placementUpdateEvent,
  * ExecuteCriticalCommandInSeparateTransaction runs a command in a separate
  * transaction that is commited right away. This is useful for things that you
  * don't want to rollback when the current transaction is rolled back.
+ * Set true to 'useExclusiveTransactionBlock' to initiate a BEGIN and COMMIT statements.
  */
 void
-ExecuteRebalancerCommandInSeparateTransaction(char *command)
+ExecuteRebalancerCommandInSeparateTransaction(char *command, bool
+											  useExclusiveTransactionBlock)
 {
 	int connectionFlag = FORCE_NEW_CONNECTION;
 	MultiConnection *connection = GetNodeConnection(connectionFlag, LocalHostName,
 													PostPortNumber);
 	StringInfo setApplicationName = makeStringInfo();
-	appendStringInfo(setApplicationName, "SET application_name TO %s",
+	appendStringInfo(setApplicationName, "SET LOCAL application_name TO %s;",
 					 CITUS_REBALANCER_NAME);
 
+	if (PropagateSessionSettingsForLoopbackConnection)
+	{
+		StringInfo setStatements = GetSetStatementsForNewConnections();
+		appendStringInfoString(setApplicationName, setStatements->data);
+	}
+
+	if (useExclusiveTransactionBlock)
+	{
+		ExecuteCriticalRemoteCommand(connection, "BEGIN;");
+	}
+
 	ExecuteCriticalRemoteCommand(connection, setApplicationName->data);
 	ExecuteCriticalRemoteCommand(connection, command);
 
+	if (useExclusiveTransactionBlock)
+	{
+		ExecuteCriticalRemoteCommand(connection, "COMMIT;");
+	}
+
 	CloseConnection(connection);
 }
 
 
+/*
+ * GetSetStatementsForNewConnections returns a string of SET statements to
+ * be executed in new connections to worker nodes.
+ */
+static StringInfo
+GetSetStatementsForNewConnections(void)
```

Comment:
nit: In general we try to avoid returning StringInfo and instead return `* char`